Geometry/GEMGeometryBuilder/src/GEMGeometryParsFromDD.cc

```diff
@@ -90,44 +95,48 @@ void GEMGeometryParsFromDD::buildGeometry(DDFilteredView& fv,
   while (doSuper) {
     // getting chamber id from eta partitions
     fv.firstChild();
-    fv.firstChild();
-    GEMDetId detIdCh =
-        GEMDetId(gemNumbering.baseNumberToUnitNumber(muonDDDNumbering.geoHistoryToBaseNumber(fv.geoHistory())));
-    // back to chambers
-    fv.parent();
-    fv.parent();
-
-    // currently there is no superchamber in the geometry
-    // only 2 chambers are present separated by a gap.
-    // making superchamber out of the first chamber layer including the gap between chambers
-
-    // In Run 3 we also have a single GE2/1 chamber at layer 2. We
-    // make sure the superchamber gets built but also we build on the
-    // first layer for the other stations so the superchamber is in
-    // the right position there.
-    if ((detIdCh.layer() == 1) || (detIdCh.layer() == 2 and detIdCh.station() == 2 and demonstratorGeometry)) {
-      buildSuperChamber(fv, detIdCh, rgeo);
-    }
-    buildChamber(fv, detIdCh, rgeo);
-
-    // loop over chambers
-    // only 1 chamber
-    bool doChambers = fv.firstChild();
-    while (doChambers) {
-      // loop over GEMEtaPartitions
-      bool doEtaPart = fv.firstChild();
-      while (doEtaPart) {
-        GEMDetId detId =
-            GEMDetId(gemNumbering.baseNumberToUnitNumber(muonDDDNumbering.geoHistoryToBaseNumber(fv.geoHistory())));
-        buildEtaPartition(fv, detId, rgeo);
+    doSuper = fv.firstChild();
+    if (doSuper) {
+      GEMDetId detIdCh =
+          GEMDetId(gemNumbering.baseNumberToUnitNumber(muonDDDNumbering.geoHistoryToBaseNumber(fv.geoHistory())));
```

Comment:
It's minor comment on the code style. On new GE2/1, we define `rawidCh` first, then use it once. In this part, we don't define new parameter, we directly get `detIdCh`.